hack/local-up-cluster.sh

```diff
@@ -379,10 +379,12 @@ cleanup()
   # Check if the proxy is still running
   [[ -n "${PROXY_PID-}" ]] && PROXY_PIDS=$(pgrep -P ${PROXY_PID} ; ps -o pid= -p ${PROXY_PID})
   [[ -n "${PROXY_PIDS-}" ]] && sudo kill ${PROXY_PIDS}
+  rm -f /tmp/kube-proxy.yaml || :
```

Comment:
we don't need `|| :` as this should never fail with the -f flag

Reply:
It could fail if user don't have permissions for removal.

I agree that it's a rare case but anyway, if we use `set -e` we should write all the code in a special way that is aware of for non-zero statuses.

Reply:
P.S. proof just in case:
```shell
$  rm -f /etc/passwd || echo rc=$?
rm: cannot remove '/etc/passwd': Permission denied
rc=1
```

Reply:
if we don't have write permissions to the temp dir, then this file would never get created. we're only looking to remove files that were created by this script. there are other instances of rm where we aren't doing this. I could see the argument to go through the script and change all instances at once, but I don't think we should do something different for these two instances when it's not done in the rest of the script